nav2_bt_navigator/behavior_trees/follow_waypoints.xml

```diff
@@ -0,0 +1,23 @@
+
+<!--
+  This Behavior Tree processes each of the given waypoints with NavigateToPose.
+-->
+<root main_tree_to_execute="MainTree">
+  <BehaviorTree ID="MainTree">
+    <ReactiveSequence name="FollowWaypoints">
+      <!-- set current goal to goals[current_waypoint_idx]. Like ComputePathToPose and FollowPath use `path_updated`, we use `goal_updated` to notify you that a goal has been updated -->
+      <SetCurrentGoal goals="{goals}" current_waypoint_idx="{current_waypoint_idx}" goal="{goal}" />
+      <NavigateToPose goal="{goal}" />
+      <!-- Return success if all goals are achieved, or update the `goal_updated` and `goal`, and then make it Running. -->
+      <ReactiveFallback name="CheckAndUpdateWaypointIndex">
+        <BlackboardCheckInt value_A="{current_waypoint_idx}" value_B="{max_waypoint_idx}" />
+        <Sequence>
+          <AddInt in_value_A="{current_waypoint_idx}" in_value_B="1" out_value="{current_waypoint_idx}" />
+          <KeepRunningUntilFailure>
+            <AlwaysSuccess />
+          </KeepRunningUntilFailure>
```

Comment:
May I know the intention of this part?

Reply:
I want `AlwaysRunning`.
I think it's necessary to ensure that BT doesn't end when the goal is updated.

Reply:
If you don't know about KeepRunningUntilFailure, please check https://github.com/BehaviorTree/BehaviorTree.CPP/pull/192.

Reply:
I'm not familiar with BT design, so any advice would be appreciated.

Reply:
I implemented `KeepRunningUntilFailure` :)

My comment is for avoiding hacks in BTs.